### PR TITLE
Add PML log entries via custom user configs

### PR DESCRIPTION
**Pull request by zamoose**
PML allows for configs in `USER_CONFIGURATION_DIR`, which defaults to `[PML install dir]/custom.user.d/` (see the [docs](http://pimpmylog.com/documentation/configuration.html)). Properly-formatted JSON files in that location will be parsed by PML.

Wrapping this up as a part of the listener role should allow us to support arbitrary custom domains.

Addresses #70.

* [x] @zamoose
* [x] @markkelnar

### CODE REVIEW

**Comment by zamoose** [comment]
Supersedes #93 with a cleaner commit history.

File: provisioning/roles/listener/meta/main.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 ---
 dependencies:
     - { role: nginx }
+    - { role: admin }
```
Comment (markkelnar) [inline provisioning/roles/listener/meta/main.yml]: What in the admin role makes the listener role dependent on it?

File: provisioning/roles/listener/templates/pimpmylog.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
+{"{{ domain }}1": {
+        "display" : "{{ enviro }} {{ backend }} Access Log",
+        "path"    : "\/var\/log\/nginx\/{{ enviro }}-{{ backend }}.apachestyle.access.log",
```
Comment (markkelnar) [inline provisioning/roles/listener/templates/pimpmylog.json]: Oh, I see the use of the variables here that map to what's passed to the listener role.  Ah, ok.

**Comment by markkelnar** [comment]
Since the admin role is not envoked by the listener role dependency, should - admin be removed from under the playbook.yml roles?

There is more in the admin role, other tham PML, that is not dependent on the listener role, ie. the stuff in roles/admin/tasks/main.yml.  Is there a problem if those tasks are run for each listener role?  Will it slow things down?  Do we rearrange things over time to separate these?

File: provisioning/roles/listener/tasks/main.yml [hunk 3]
```diff
@@ -5,3 +5,6 @@
 - name: "Set up cache alias for {{ enviro }} {{ backend }}"
   template: src=etc/nginx/wpcache.conf dest=/etc/nginx/conf.d/cache.{{ enviro }}-{{ backend }}.conf owner=root group=root mode=0644
   notify: nginx reload
+
+- name: "Enable PML viewing for {{ enviro }} {{ backend }}"
+  template: src=pimpmylog.json dest={{ wp_doc_root }}/admin/logs/config.user.d/{{ enviro }}-{{ backend }}.json
```
Comment (markkelnar) [inline provisioning/roles/listener/tasks/main.yml]: Pretty cool use here for each backend to reduce the template size!

**Comment by zamoose** [comment]
I added admin as a dependency so that we can make sure the PML directories are created correctly before trying to put JSON files in them.

**Comment by markkelnar** [comment]
@zamoose A merge conflict now exists, but looks good otherwise.

**Comment by markkelnar** [comment]
It's conflicting with the mailhog tasks added to the bottom of admin/tasks/main.yml.

**Pull request merged**